services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/map/ProcessedMapRepositoryImpl.java

```diff
@@ -0,0 +1,300 @@
+                try {
+                    Document document = collection.find(eq("properties.intersectionId", intersectionId))
+                        .projection(projectionFields).sort(Sorts.descending("properties.timeStamp")).maxTime(props.getMongoTimeoutMs(), TimeUnit.MILLISECONDS).first();
+                
+                    if(document != null){
+                        IntersectionReferenceData data = new IntersectionReferenceData();
+                        Document properties = document.get("properties", Document.class);
+
+                        if (properties != null) {
+                            Document refPoint = properties.get("refPoint", Document.class);
+                            data.setIntersectionID(intersectionId);
+                            data.setRoadRegulatorID("-1");
+                            data.setRsuIP(properties.getString("originIp"));
+
+                            if(properties.getString("intersectionName") != null && properties.getString("intersectionName").isEmpty()){
+                                data.setIntersectionName(properties.getString("intersectionName"));
+                            }
+                            
+                            if (refPoint != null) {
+                                data.setLatitude(refPoint.getDouble("latitude"));
+                                data.setLongitude(refPoint.getDouble("longitude"));
+                            }
+                        }
+                        referenceDataList.add(data);
+                    }
+                } catch (MongoException e){
+                    logger.error("MongoDB Intersection Query Did not finish in allowed time window");
+                } catch (Exception e) {
+                    logger.error(e.getMessage());
+                }
+                
+            }
+        }
+
+        return referenceDataList;
+    }
+
+    public List<IntersectionReferenceData> getIntersectionsContainingPoint(double longitude, double latitude){
+        MongoCollection<Document> collection = mongoTemplate.getCollection(collectionName);
+        DistinctIterable<Integer> docs = collection.distinct("properties.intersectionId", Integer.class);
+        MongoCursor<Integer> results = docs.iterator();
+        MapIndex index = new MapIndex();
+        Map<Integer, ProcessedMap<LineString>> mapLookup = new HashMap<>();
+        while (results.hasNext()) {
+            Integer intersectionId = results.next();
+            if (intersectionId != null){
+
+                
+                Query query = getQuery(intersectionId,  null,  null,  true,  true);
+
+                List<ProcessedMap<LineString>> maps = findProcessedMaps(query);
+
+                if(maps.size() > 0){
+                    MapBoundingBox box = new MapBoundingBox(maps.getFirst());
+                    index.insert(box);
+                    mapLookup.put(intersectionId, maps.getFirst());
+                }
+            }
+        }
```

Comment:
issue(blocking): use try-with-resources to ensure the auto closeable lifecycle is managed correctly. The current implementation won't close the MongoCursor iterator in the event of an exception so resource leakage can occur
```suggestion
        MapIndex index;
        Map<Integer, ProcessedMap<LineString>> mapLookup;
        try (MongoCursor<Integer> results = docs.iterator()) {
            index = new MapIndex();
            mapLookup = new HashMap<>();
            while (results.hasNext()) {
                Integer intersectionId = results.next();
                Query query = getQuery(intersectionId, null, null, true, true);

                List<ProcessedMap<LineString>> maps = findProcessedMaps(query);

                if (!maps.isEmpty()) {
                    MapBoundingBox box = new MapBoundingBox(maps.getFirst());
                    index.insert(box);
                    mapLookup.put(intersectionId, maps.getFirst());
                }
            }
        }
```